crates/sui-orchestrator/src/measurement.rs

```diff
@@ -0,0 +1,396 @@
+pub struct Measurement {
+    /// Duration since the beginning of the benchmark.
+    timestamp: Duration,
+    /// Latency buckets.
+    buckets: HashMap<BucketId, usize>,
+    /// Sum of the latencies of all finalized transactions.
+    sum: Duration,
+    /// Total number of finalized transactions
+    count: usize,
+    /// Square of the latencies of all finalized transactions.
+    squared_sum: Duration,
+}
+
+impl Measurement {
+    // Make a new measurement from the text exposed by prometheus.
+    pub fn from_prometheus(text: &str) -> Self {
+        let br = std::io::BufReader::new(text.as_bytes());
+        let parsed = Scrape::parse(br.lines()).unwrap();
+
+        let buckets: HashMap<_, _> = parsed
+            .samples
+            .iter()
+            .find(|x| x.metric == "latency_s")
+            .map(|x| match &x.value {
+                prometheus_parse::Value::Histogram(values) => values
+                    .iter()
+                    .map(|x| {
+                        let bucket_id = x.less_than.to_string();
+                        let count = x.count as usize;
+                        (bucket_id, count)
+                    })
+                    .collect(),
+                _ => panic!("Unexpected scraped value"),
+            })
+            .unwrap_or_default();
+
+        let sum = parsed
+            .samples
+            .iter()
+            .find(|x| x.metric == "latency_s_sum")
+            .map(|x| match x.value {
+                prometheus_parse::Value::Untyped(value) => Duration::from_secs_f64(value),
+                _ => panic!("Unexpected scraped value"),
+            })
+            .unwrap_or_default();
+
+        let count = parsed
+            .samples
+            .iter()
+            .find(|x| x.metric == "latency_s_count")
+            .map(|x| match x.value {
+                prometheus_parse::Value::Untyped(value) => value as usize,
+                _ => panic!("Unexpected scraped value"),
+            })
+            .unwrap_or_default();
+
+        let squared_sum = parsed
+            .samples
+            .iter()
+            .find(|x| x.metric == "latency_squared_s")
+            .map(|x| match x.value {
+                prometheus_parse::Value::Counter(value) => Duration::from_secs_f64(value),
+                _ => panic!("Unexpected scraped value"),
+            })
+            .unwrap_or_default();
+
+        let timestamp = parsed
+            .samples
+            .iter()
+            .find(|x| x.metric == "benchmark_duration")
+            .map(|x| match x.value {
+                prometheus_parse::Value::Counter(value) => Duration::from_secs(value as u64),
+                _ => panic!("Unexpected scraped value"),
+            })
+            .unwrap_or_default();
+
+        Self {
+            timestamp,
+            buckets,
+            sum,
+            count,
+            squared_sum,
+        }
+    }
+
+    /// Compute the tps.
+    pub fn tps(&self) -> u64 {
```

Comment:
how long are experiments run for? could we be paying a warmup cost of low tps here that we don't see since we take the average over the experiment?

Reply:
I see that you run multiple experiments to find the breaking point on the same infra, are we sure that the system has processed everything from the previous experiment and there is no interference?

Reply:
You choose the duration through the command line arguments (default is 3 minutes). We may be paying a warmup cost. To measure that, we will modify the plot scripts to create a new plot: TPS vs Time (and Latency vs Time).